### PR TITLE
AUTO-639 added optional 'hdd_size' yaml value to change HDD size for autoscaling units

### PR DESCRIPTION
**Pull request by snayler0**
build passing: https://gaautobots.atlassian.net/builds/browse/AM-AUT24-3

### CODE REVIEW

File: amazonia/classes/asg.py [hunk 1]
```diff
@@ -62,7 +65,7 @@ def __init__(self, title, vpc, template, minsize, maxsize, subnets, load_balance
 
     def create_asg(self, title, minsize, maxsize, subnets, load_balancer, keypair, image_id, instance_type,
                    health_check_grace_period, health_check_type, iam_instance_profile_arn, userdata, sns_topic_arn,
-                   sns_notification_types):
+                   sns_notification_types, hdd_size):
```
Comment (fuddstar) [inline amazonia/classes/asg.py]: Sorry I know its not your code, but could you update the link in the docstring from http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-security-group.html  to http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-as-group.html

File: amazonia/classes/asg.py [hunk 2]
```diff
@@ -145,6 +151,19 @@ def create_launch_config(self, title, keypair, image_id, instance_type, iam_inst
...
+                    Ebs=ec2.EBSBlockDevice(
+                        VolumeSize=hdd_size,
+                        VolumeType='gp2'
+                    )
+                ),
+            ]
+
         return launch_config_title
 
     def create_cd_deploygroup(self, title, cd_service_role_arn):
```
Comment (fuddstar) [inline amazonia/classes/asg.py]: Also update this one to http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codedeploy-application.html
and add http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codedeploy-deploymentgroup.html

File: amazonia/classes/asg.py [hunk 3]
```diff
@@ -111,13 +115,14 @@ def create_asg(self, title, minsize, maxsize, subnets, load_balancer, keypair, i
...
-            userdata=userdata
+            userdata=userdata,
+            hdd_size=hdd_size
         ))
         if userdata is None:
             self.lc.UserData = ''
         return title
 
-    def create_launch_config(self, title, keypair, image_id, instance_type, iam_instance_profile_arn, userdata):
+    def create_launch_config(self, title, keypair, image_id, instance_type, iam_instance_profile_arn, userdata, hdd_size):
```
Comment (fuddstar) [inline amazonia/classes/asg.py]: Update
add http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-as-launchconfig.html
and add http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-as-launchconfig-blockdev-mapping.html

File: web/web_app.yaml [hunk 4]
```diff
@@ -3,6 +3,7 @@ autoscaling_units:
   -
     unit_title: 'api'
     image_id: 'ami-dc361ebf'
+    hdd_size: '20'
```
Comment (fuddstar) [inline web/web_app.yaml]: Please remove this file

File: amazonia/classes/asg.py (hunk 1 above)
Comment (snayler0) [inline amazonia/classes/asg.py]: Done :)

File: amazonia/classes/asg.py (hunk 2 above)
Comment (snayler0) [inline amazonia/classes/asg.py]: Done :)

File: amazonia/classes/asg.py (hunk 3 above)
Comment (snayler0) [inline amazonia/classes/asg.py]: Done :)

File: web/web_app.yaml (hunk 4 above)
Comment (snayler0) [inline web/web_app.yaml]: Removed the hdd_size line. I think that's what you meant? :)

**Comment by fuddstar** [comment]
:shipit:

**Pull request merged**